### ISSUE TITLE
A11y_AzureToolsStorageExplorer_SASGenerate_Keyboard: Keyboard focus is not in logical order after Close control.

### ISSUE THREAD

**Issue by Suraj1200499** [microsoft/AzureStorageExplorer#1912]
“[Check out Accessibility Insights!](https://accessibilityinsights.io/) Identify accessibility bugs before check-in and make bug fixing faster and easier.”

GitHubTags: #A11y_AzureToolsStorageExplorer; #A11yMAS; #A11YTCS; #DesktopApp; #MAS2.4.3; #Keyboard; #Win32; 

**Environment Details:**
Application Name: Azure Tools Storage Explorer
Application Version: 1.10.0
Windows10

**Repro Steps:**
1. Launch Storage Explorer
2. Navigate to "Open Connect dialog" pane and Sign in to Azure.
3. Go to "Manage Accounts" pane and Select the subscriptions you will use. Once selected, click on "Apply" button.
4. Navigate to "Blob Containers" of the selected subscription, right click and hit enter on "Create Blob container" to Create a new Blob container.
5. Once completed, Navigate to "Upload" button and Upload a blob/file by using  "Upload Files" control.
6. Blob gets uploaded, right click on blob and select "Get Shared Access Signature" control and hit enter.
7. Navigate to "Create" button and hit enter.
8. Start navigating on the screen using Tab key.
9. Verify Keyboard focus is not in logical order after Close control.

**​Actual:​**
While navigating on the screen using Tab key, Keyboard focus is not in logical order after Close control.

**Expected:**
While navigating on the screen using Tab key, Keyboard focus should be in logical order after Close control.
Means after close control when we tab keyboard focus should shift to copy button of "Url" disabled edit box, focus should not shift to "Query string" disabled edit box.
And also keyboard focus should not shift to disabled controls while navigating on the screen using Tab key.

**UserImpact:**
Keyboard only users face difficulty to navigate on the screen if keyboard focus is not in logical order on the screen.

**Recommendation:**
https://microsoft.sharepoint.com/teams/msenable/mas/Pages/browse-fixes.aspx

**MAS Reference:**
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={0de7fbe1-ad7e-48e5-bcbb-8d986691e2b9}

**Attachment for Reference:**
[A11y_AzureToolsStorageExplorer_SASURL_Keyboard.pptx](https://github.com/microsoft/AzureStorageExplorer/files/3652327/A11y_AzureToolsStorageExplorer_SASURL_Keyboard.pptx)

**Comment by craxal** [comment]
Verified. When the panel opens, the token input gets focus first. Tab then moves focus to the first control and cycles normally (including the token input) until after the Close button. There, it moves back to the token input, then to the first control, and repeats. No idea why it's doing that, though.

**Comment by Suraj1200499** [comment]
@craxal When tab to move focus from the Close button, keyboard focus shifts to "Query String" read-only control and then to "Container" control which is not logical. Although screen reader not announces this but keyboard only users face in difficulty for understanding the keyboard focus order.

So after Close button if we tab keyboard focus should shift to "Container" control instead of shifting to "Query String" control.

**Comment by craxal** [comment]
This bug should automatically resolve when we remove the extra tab stop on the dialog `<body>`/window, which is resolved as of Electron v5.0.0 (electron/electron#12919). Which means we can't fix this bug until after we update Electron.

A potential workaround is we can re-style the selected text to only show up if the input element has focus. @MRayermannMSFT, however, doesn't think this this workaround is appropriate (I'll let him leave a comment explaining why if he wants).

**Comment by Suraj1200499** [comment]
@Craxal 
As mentioned "Electron Update Required", We will verify this issue Once "Microsoft Tools Storage Explorer" application is updated to Newer Version(1.12.0).

**Comment by craxal** [comment]
Verified this is fixed after upgrading to Electron 7.x.

**Comment by Alexandria-R** [comment]
Approved for exclusion as medium impact.

**Comment by mstechie** [comment]
GitHubTags:#BM-TCS-StorageExplorer-Win32-Sep2019;
